src/thor/route_matcher.cc

```diff
@@ -288,6 +330,7 @@ bool RouteMatcher::FormPath(const std::shared_ptr<DynamicCost>* mode_costing,
     float de_remaining_length = de->length() * (1 - edge.percent_along());
     float de_length = length_comparison(de_remaining_length, true);
     EdgeLabel prev_edge_label;
+    Cost elapsed;
```

Comment:
the elapsed cost is in the inner loop because the outer loop is in case a potential edge candidate fails. again we dont want to count that failures elapsed time toward the overall correct match so we reset for each candidate path we try